File: 0605-can-place-flowers/0605-can-place-flowers.js
/**
 * @param {number[]} flowerbed
 * @param {number} n
 * @return {boolean}
 */
var canPlaceFlowers = function(flowerbed, n) {
    let plantCnt = 0;
    if(!flowerbed[0] && !flowerbed[1]) {
        plantCnt += 1;
        flowerbed[0] = 1;
    }
    for(let i = 1; i < flowerbed.length - 1; i++) {
        const [l, m, r] = [flowerbed[i - 1], flowerbed[i], flowerbed[i + 1]];
        if(!l && !m && !r) {
            plantCnt += 1;
            flowerbed[i] = 1;
        }
    }
    if(!flowerbed[flowerbed.length - 2] && !flowerbed[flowerbed.length - 1]) {
        plantCnt += 1;
    }
    return plantCnt >= n ? true : false; 
};

File: 0605-can-place-flowers/0605-can-place-flowers.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        if(n == 0) return true;
        int plantCnt = 0;
        for(int i = 0; i < flowerbed.length; i++) {
            if(flowerbed[i] == 0 && checkLeft(flowerbed, i) && checkRight(flowerbed, i)) {
                flowerbed[i] = 1;
                plantCnt += 1;
            }

            if(plantCnt == n) return true;
        }

        return false;
    }

    public boolean checkLeft(int[] flowerbed, int i) {
        if(i == 0 || flowerbed[i - 1] == 0) return true;
        return false;
    }

    public boolean checkRight(int[] flowerbed, int i) {
        if(i == flowerbed.length - 1 || flowerbed[i + 1] == 0) return true;
        return false;
    }
}